src/pages/workspaces/workspace/analysis/modals/AnalysisModal.js

```diff
@@ -96,16 +99,25 @@ export const AnalysisModal = withDisplayName('AnalysisModal')(
       [environmentMode, getEnvironmentView],
       [Utils.DEFAULT, renderSelectAnalysisBody])
 
-    const getEnvironmentView = () => Utils.switchCase(currentTool,
-      [tools.Jupyter.label, renderComputeModal],
-      [tools.RStudio.label, renderComputeModal],
-      [tools.Galaxy.label, () => renderAppModal(GalaxyModalBase, tools.Galaxy.label)],
-      [tools.Cromwell.label, () => renderAppModal(CromwellModalBase, tools.Cromwell.label)],
-      [tools.Azure.label, renderAzureModal]
+    const getEnvironmentView = () => Utils.switchCase(cloudProvider,
+      [cloudProviderTypes.GCP, getGCPEnvironmentView],
+      [cloudProviderTypes.AZURE, getAzureEnvironmentView]
+    )
+
+    const getGCPEnvironmentView = () => Utils.switchCase(currentTool,
+      [toolLabels.Jupyter, renderComputeModal],
+      [toolLabels.RStudio, renderComputeModal],
+      [toolLabels.Galaxy, () => renderAppModal(GalaxyModalBase, toolLabels.Galaxy)],
+      [toolLabels.Cromwell, () => renderAppModal(CromwellModalBase, toolLabels.Cromwell)],
+      [toolLabels.JupyterLab, renderAzureModal]
```

Comment:
This case isn't possible is it? Just want to make sure I'm not missing something because this is going to change in my upcoming JupyterLab GCP PR

Reply:
That's a good catch! That case was not possible in my branch. Removing the JupyterLab line from my branch - yours will add it back in correctly.